raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -171,6 +171,12 @@ private boolean updateHighWatermark() {
                         || (highWatermarkUpdateOffset == currentHighWatermarkMetadata.offset &&
                             !highWatermarkUpdateMetadata.metadata.equals(currentHighWatermarkMetadata.metadata))) {
                         highWatermark = highWatermarkUpdateOpt;
+                        log.debug(
```

Comment:
Do you think this level of detail is more suitable for trace? We have debug logging for high watermark advances in `KafkaRaftClient`.

Reply:
Yes. Updated the PR to use trace.